Add explanation of how to change allowed inbound IP addresses (#1484)
alan-turing-institute/data-safe-haven: docs/source/roles/system_manager/manage_users.md

>
> --details about network and location/VPN restrictions here--

(modifying_network)=

> ## {{globe_with_meridians}} Manually modifying network restrictions

Reviewer: _If_ this is going to be under managing users, it might be better to title this something like "Changing access restrictions" to clarify how it relates to users.

Author: How about "Changing user network access restrictions", making it clear that this means changing a network setting rather than any other kind of access restriction (such as disabling MFA)
```suggestion
## {{globe_with_meridians}} Changing user network access restrictions
```

Reviewer: Perfect :+1:.

> One of the controls used by Tier 2/3 SREs is to restrict access based on network addresses. If the users were to require access from different IP addresses - if their institutional IP address changes, or an additional location with a different network were approved - then the network addresses that are allowed access can be modified manually.

Reviewer: ```suggestion
One of the controls used by Tier 2/3 SREs is to restrict access based on network addresses. 
The network addresses that are allowed to access an SRE can be modified after deployment.
This is useful if users require access from new, or different, IP addresses.
For example if their institutional IP address changes, or an additional location is approved.
```

- In the Azure Portal, navigate to `RG_SHM_<SHM_ID>_SRE_<SRE_ID>_NETWORKING`

> - On the `Overview` tab, navigate to the Network Security Group for the appropriate remote desktop service (e.g `NSG_SHM_<SHM_ID>_SRE_<SRE_ID>_GUACAMOLE`)

Reviewer: ```suggestion
- On the `Overview` tab, navigate to the Network Security Group for the appropriate remote desktop service (e.g. `NSG_SHM_<SHM_ID>_SRE_<SRE_ID>_GUACAMOLE`)
```

> - On the `Overview` tab, navigate to the Network Security Group for the appropriate remote desktop service (e.g `NSG_SHM_<SHM_ID>_SRE_<SRE_ID>_GUACAMOLE`)

Reviewer: Is `NSG_SHM_<SHM_ID>_SRE_<SRE_ID>_GUACAMOLE` an example, or is that always the form of the NSG's name? (After removing MSRDS)

Author: It'll always be the form of the NSG's name after removing MSRDS. I left it at this in case we want this to go in `release-4.0.4`, but if not, can modify it.

Reviewer: I think we can cross that bridge when/if we come to it.

There is no sense in making the develop branch 'worse' or 'incorrect' in anticipation of merging the changes somewhere else later.

Author: ```suggestion
- On the `Overview` tab, navigate to `NSG_SHM_<SHM_ID>_SRE_<SRE_ID>_GUACAMOLE`, the Network Security Group for the remote desktop service
```

- Navigate to `Inbound Security Rules`, and open the entry called `AllowUsersApprovedHttpsInbound`
- For the field `Source IP addresses/CIDR ranges`, add the desired IP address or range to the existing entry, or overwrite the existing entry if it is no longer required

> - Users will then be able to access the remote desktop interface from the new IP address

Reviewer: It might be worth rephrasing these to say something like "Update the Source IP addresses/CIDR ranges field to include the IP addresses (and only the IP addresses) that should be able to access the SRE" and "Users will now be able to access the remote desktop from only the desired IP addresses".

That way we emphasise that this should _only_ contain approved IP addresses, rather than being a list which we repeatedly append to.

Author: Rather spell it out a bit more than say "include only IP addresses", as that implies there are IP addresses that need to be removed, which may not be the case. 

```suggestion
- Update the `Source IP addresses/CIDR ranges` field to include IP addresses that should be able to access the SRE and remove any that should not
- Users will now be able to access the remote desktop interface from only the desired IP addresses
```

Reviewer: Sounds good to me. Both are equivalent but we should go for whatever is clearer. I think you are right that the process here is 'making a change' so it makes sense to present that as add/remove rather than rewrite from scratch.

(This is kind of the same as the declarative vs imperative)
